### PR TITLE
Restructure event bus to allow them to control message commit behavior

### PR DESCRIPTION
**Pull request by morsecodist**
This allows event bus implementations to decide whether or not to commit messages.

### CODE REVIEW

File: workflows/plugins/event_bus/redis/event_bus_redis.py [hunk 1]
```diff
@@ -12,12 +13,17 @@
...
     def __init__(self, setings: RedisEventBusSettings) -> None:
-        self.settings = setings
-        self.redis = aioredis.from_url(self.settings.REDIS_URL)
+        self._settings = setings
+        self._redis = aioredis.from_url(self._settings.REDIS_URL)
+        self._logger = logging.getLogger("EventBusRedis")
 
     async def send(self, message: WorkflowStatusMessage) -> None:
-        await self.redis.lpush(self.settings.QUEUE_NAME, message.model_dump_json())
+        await self._redis.lpush(self.settings.QUEUE_NAME, message.model_dump_json())  # type: ignore
```
Comment (rzlim08) [inline workflows/plugins/event_bus/redis/event_bus_redis.py:21]: change to self._settings

File: workflows/api/loader/loader.py [hunk 2]
```diff
@@ -67,62 +74,71 @@ async def process_workflow_completed(
...
-                        .options(
-                            joinedload(WorkflowRun.workflow_version).joinedload(WorkflowVersion.workflow),
-                            selectinload(WorkflowRun.entity_inputs),
+    def _bind_handle_message(self) -> Callable[[WorkflowStatusMessage], Awaitable[None]]:
+        async def handle_message(event: WorkflowStatusMessage) -> None:
+            print("event", event, file=sys.stderr)
+            if isinstance(event, WorkflowStartedMessage):
+                workflow_run = (
+                    await self.session.execute(select(WorkflowRun).where(WorkflowRun.execution_id == event.runner_id))
+                ).scalar_one_or_none()
```
Comment (rzlim08) [inline workflows/api/loader/loader.py:83]: Right now with this one_or_none logic, if we don't find a matching workflow run we just ignore it and silently drop the message. Should we be raising an error here?

Comment (morsecodist) [reply workflows/api/loader/loader.py:83]: No, this is intentional because we will be getting mngs messages

Comment (rzlim08) [reply workflows/api/loader/loader.py:83]: ok makes sense, so the retry only happens if there is an error in the loaders

Comment (morsecodist) [reply workflows/api/loader/loader.py:83]: If there is an error in the loaders or if for whatever reason the process is randomly killed. Loader errors may not really be retry-able unless there is a transient issue communicating with the entities service so for those failures it is more about pushing them into the dead letter queue than retrying.

**Review by rzlim08 — APPROVED** [review]
Looks good, just one small nit left

**Pull request merged**